src/modules/tools/reporter/Reporter.cpp

```diff
@@ -0,0 +1,147 @@
+                              //C=running config file, A=paused, D=pausing, R=resuming, B=busy (running a macro)
+                              gcode->stream->printf("{\"status\":\"");
+                              if(THEKERNEL->conveyor->is_queue_empty()){
+                                gcode->stream->printf("I");
+                              }else{
+                                gcode->stream->printf("P");
+                              }
+                              if (THEKERNEL->is_halted()){
+                                gcode->stream->printf("S");
+                              }
+                              void *returned_data;
+                              if (PublicData::get_value( player_checksum, is_suspended_checksum, &returned_data )) {
+                                bool b = *static_cast<bool *>(returned_data);
+                                if(b){
+                                  gcode->stream->printf("A");
+                                }
+                              }
+                              gcode->stream->printf("\",\"heaters{{\":[");
+                              std::vector<struct pad_temperature> controllers;
+                              if (PublicData::get_value(temperature_control_checksum, poll_controls_checksum, &controllers)) {
+                                for (auto &c : controllers) {
+                                  if(&c == &controllers.back()){
+                                    printf("%f,", c.current_temperature);
+                                  }else{
+                                    printf("%f,", c.current_temperature);
+                                  }
+                                }
+
+                                gcode->stream->printf("],\"active\":[");
+                                for (auto &c : controllers) {
+                                  if(&c == &controllers.back()){
+                                    printf("%f", c.target_temperature);
+                                  }else{
+                                    printf("%f,", c.target_temperature);
+                                  }
+                                }
+
+                              // Smoothieware has no concept of a standby temperature.
+                              gcode->stream->printf("],\"standby\":[0,0,0],");
+
+                              // Heater status
+                              // status of the heaters, 0=off, 1=standby, 2=active, 3=fault
+                              gcode->stream->printf("\"hstat\":[");
+                              for(auto &c : controllers){
+                                // off
+                                if(c.target_temperature == 0){
+                                  gcode->stream->printf("0");
+                                }
+                                // active
+                                if(c.target_temperature>c.current_temperature){
```

Comment:
This is not the correct way to determine it is active... if target_temperature is > 0 then it is active it has nothing to do with the current_temperature at all.